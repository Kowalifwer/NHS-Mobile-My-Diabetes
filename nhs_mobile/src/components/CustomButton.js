import React from 'react';
import {
    Pressable,
    Text,
    StyleSheet,
} from 'react-native';

const CustomButton = (props) => {
    return (
        <Pressable
            onPress={props.onPressFunction}
            hitSlop={{ top: 10, bottom: 10, right: 10, left: 10 }}
            android_ripple={{ color: '#00000050' }}
            style={({ pressed }) => [
                { backgroundColor: pressed ? '#dddddd' : props.color },
                styles.button,
                { ...props.style }
            ]}
        >
            <Text style={styles.text}>
                {props.title}
            </Text>
        </Pressable>
    )
}

const styles = StyleSheet.create({
    text: {
        color: '#ffffff',
        fontSize: 20,
        textAlign: 'center',
        marginTop: -5,
    },
    button: {
        width: 350,
        minHeight: 40,
        height: 'auto',
        alignItems: 'center',
        justifyContent: 'center',
        borderRadius: 10,
        margin: 10,
        padding: 10,
    },
})

export default CustomButton;