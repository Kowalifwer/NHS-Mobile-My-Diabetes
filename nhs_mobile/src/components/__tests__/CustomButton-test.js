import React from 'react';
import renderer from 'react-test-renderer';
import CustomButton from '../CustomButton';

it('renders correctly across screens', () => {
    const tree = renderer.create(<CustomButton />).toJSON();
    expect(tree).toMatchSnapshot();
  });