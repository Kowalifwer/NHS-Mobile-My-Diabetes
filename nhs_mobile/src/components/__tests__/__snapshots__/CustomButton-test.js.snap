// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly across screens 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  hitSlop={
    Object {
      "bottom": 10,
      "left": 10,
      "right": 10,
      "top": 10,
    }
  }
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": undefined,
      },
      Object {
        "alignItems": "center",
        "borderRadius": 10,
        "height": "auto",
        "justifyContent": "center",
        "margin": 10,
        "minHeight": 40,
        "padding": 10,
        "width": 350,
      },
      Object {},
    ]
  }
>
  <Text
    style={
      Object {
        "color": "#ffffff",
        "fontSize": 20,
        "marginTop": -5,
        "textAlign": "center",
      }
    }
  />
</View>
`;
